spring-boot/src/main/java/org/springframework/boot/ResourceBanner.java

```diff
@@ -48,21 +52,25 @@
 
 	private static final Log logger = LogFactory.getLog(ResourceBanner.class);
 
-	private Resource resource;
+	private Resource bannerResource;
+	private Resource gitPropertiesResource;
+	private Resource buildInfoPropertiesResource;
+
+	public ResourceBanner(Resource bannerResource, Resource gitPropertiesResource, Resource buildInfoPropertiesResource) {
```

Comment:
Since it's a public class with new optional fields, keep the original Constructor and delegate to this one.